pkl-core/src/main/java/org/pkl/core/ast/member/ListingOrMappingTypeCastNode.java

```diff
@@ -54,6 +73,20 @@ public SourceSection getSourceSection() {
 
   @Override
   public Object execute(VirtualFrame frame) {
+    if (owner == null || receiver == null) return executeInternal(frame);
+    var prevOwner = VmUtils.getOwner(frame);
+    var prevReceiver = VmUtils.getReceiver(frame);
+    try {
+      VmUtils.setOwner(frame, owner);
+      VmUtils.setReceiver(frame, receiver);
+      return executeInternal(frame);
+    } finally {
+      VmUtils.setOwner(frame, prevOwner);
+      VmUtils.setReceiver(frame, prevReceiver);
+    }
+  }
+
+  private Object executeInternal(VirtualFrame frame) {
```

Comment:
mild preference to just inline, myself
```suggestion
      return typeNode.execute(frame, frame.getArguments()[2]);
    } catch (VmTypeMismatchException e) {
      CompilerDirectives.transferToInterpreter();
      throw e.toVmException();
    } finally {
      VmUtils.setOwner(frame, prevOwner);
      VmUtils.setReceiver(frame, prevReceiver);
    }
  }
```

(GH suggestion can't span out of the diff, alas)

Reply:
Inlining would make it slower, right? I'd always have to get and reset the owner/receiver in the frame, while right now it's short-circuiting for the 99% case where it doesn't come from a typealias.

Reply:
You are entirely right... I had missed that top line somehow